src/basic/process-util.c

```diff
@@ -2262,3 +2267,42 @@ _noreturn_ void report_errno_and_exit(int errno_fd, int error) {
 
         _exit(EXIT_FAILURE);
 }
+
+int getpidfdid_cached(uint64_t *ret) {
```

Comment:
So I've extracted a bunch of pidfd helpers into its own pidfd.[ch] in #35242. Maybe want to rebase on that later?